### PR TITLE
[android][utils] Support ChannelsLast in TensorImageUtils

### PR DESCRIPTION
**Pull request by IvanKobzarev**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#48990 [android][utils] Support ChannelsLast in TensorImageUtils**

Introducing TensorImageUtils methods to prepare tensors in channelsLast MemoryFormat.
ChannlesLast is preferred for performance (For resnet18 it is about -10% on Pixel 3)

Not to introduce api breaking changes, adding additional parameter MemoryFormat which is CONTIGUOUS by default.

Testing by checking test_app that uses this call
```
gradle -p android test_app:installMnetLocalCameraDebug -PABI_FILTERS=arm64-v8a
```

Differential Revision: [D27173940](https://our.internmc.facebook.com/intern/diff/D27173940)

### CODE REVIEW

**Comment by jeffxtang** [comment]
Went through the TestApp build, install and perf comparison test process (using both resnet19 and mnetv3) and saw 10%-33% perf improvement (forward time). All looks good..

**Comment by jeffxtang** [comment]
Went through the TestApp build, install and perf comparison test process (using both resnet19 and mnetv3) and saw 10%-33% perf improvement (forward time). All looks good.

File: android/pytorch_android_torchvision/src/main/cpp/pytorch_vision_jni.cpp [hunk 1]
```diff
@@ -91,32 +95,64 @@ static void imageYUV420CenterCropToFloatBuffer(
...
+        yBeforeRtn = cropYAdd + cropYMult * (int)(y * scale);
+        yIdx = yBeforeRtn * yRowStride + xBeforeRtn * yPixelStride;
+        uvIdx = (yBeforeRtn >> 1) * uvRowStride + (xBeforeRtn >> 1) * uvPixelStride;
+        ui = uData[uvIdx];
+        vi = vData[uvIdx];
+        yi = yData[yIdx];
+        yi = (yi - 16) < 0 ? 0 : (yi - 16);
+        ui -= 128;
+        vi -= 128;
+        a0 = 1192 * yi;
```
Comment (jeffxtang) [inline android/pytorch_android_torchvision/src/main/cpp/pytorch_vision_jni.cpp:115]: what're these magic numbers for?

**Review by jeffxtang — APPROVED** [review]
Went through the TestApp build, install and perf comparison test process (using both resnet19 and mnetv3) and saw 10%-33% perf improvement (forward time). All looks good, but will you update the description to include comparison perf numbers for mnet3 (13ms vs 21ms) and resnet18 (200ms vs 220ms, both on Pixel 3). Also, this is my updated make_assets.py:
```
import torch
import torchvision
from torch.utils.mobile_optimizer import optimize_for_mobile

print(torch.version.__version__)
# should be 1.8 or above

resnet18 = torchvision.models.resnet18(pretrained=True)
resnet18.eval()
optimize_for_mobile(torch.jit.trace(resnet18, torch.rand(1, 3, 224, 224))).save("app/src/main/assets/resnet18.pt")

resnet50 = torchvision.models.resnet50(pretrained=True)
resnet50.eval()
optimize_for_mobile(torch.jit.trace(resnet50, torch.rand(1, 3, 224, 224))).save("app/src/main/assets/resnet50.pt")

mobilenet2q = torchvision.models.quantization.mobilenet_v2(pretrained=True, quantize=True)
mobilenet2q.eval()
optimize_for_mobile(torch.jit.trace(mobilenet2q, torch.rand(1, 3, 224, 224))).save("app/src/main/assets/mobilenet2q.pt")

mobilenet3s = torchvision.models.mobilenet_v3_small(pretrained=True)
mobilenet3s.eval()
optimize_for_mobile(torch.jit.trace(mobilenet3s, torch.rand(1, 3, 224, 224))).save("app/src/main/assets/mnet.pt")
```

File: android/pytorch_android_torchvision/src/main/cpp/pytorch_vision_jni.cpp (hunk 1 above)
Comment (IvanKobzarev) [reply android/pytorch_android_torchvision/src/main/cpp/pytorch_vision_jni.cpp:115]: This is yuv420 to rgb conversion https://en.wikipedia.org/wiki/YUV

**Pull request closed without merging**